File: lab-2/src/moves/DoubleTeam.java
package moves;

import ru.ifmo.se.pokemon.*;

public class DoubleTeam extends StatusMove {
    public DoubleTeam() {
        super(Type.NORMAL, 0, 100);
    }

    protected void applySelfEffects(Pokemon p) {
        p.setMod(Stat.EVASION, 1);
    }

    protected String describe() {
        return "is using Double Team";
    }
}

File: lab-2-kotlin/src/moves/DoubleTeam.kt
package moves

import ru.ifmo.se.pokemon.*

class DoubleTeam : StatusMove(Type.NORMAL, 0.0, 100.0) {
    override fun applySelfEffects(p: Pokemon) {
        p.setMod(Stat.EVASION, 1)
    }

    override fun describe(): String {
        return "is using Double Team"
    }
}